app/controllers/add-preprint.js

```diff
@@ -1,87 +1,207 @@
         },
-        uploadNewPreprintToNewProject(title, abstract, authors, subject, tags, journal, doi) {
-            //Create new public project
-            let node = this.get('store').createRecord('node', {
-                title: title,
+        createProject(resolve) {
+            const node = this.get('store').createRecord('node', {
+                title: this.get('nodeTitle'),
                 category: 'project',
-                description: abstract || null,
                 public: true
             });
-
-            this.set('preprint', {
-                title: title,
-                abstract: abstract,
-                authors: authors,
-                subject: subject,
-                tags: tags,
-                journal: journal,
-                doi: doi
-            });
-
-            // TODO: Add logic for if you're not uploading a file
-            // Upload a new file to the new project
             node.save().then(() => {
-                this.set('_url', `https://test-files.osf.io/file?path=/${this.get('latestFileName')}&nid=${node.id}&provider=osfstorage`);
-                this.set('openModal', false);
-                this.get('resolve')();
-            }).catch(error => {
-                console.error(error);
-                this.get('toast').error('Could not create preprint project!');
+                this.set('preprintNode', node);
+                this._refreshNodes();
+                this.send('startUpload', resolve);
             });
-            //TODO: eventually make a call to set an OSF file as a preprint (will probably need a flag for such)
+        },
+        deleteProject(resolve) {
+            if (this.get('preprintNode')) {
+                this.get('preprintNode').destroyRecord().then(() => {
+                    this.get('toast').info('Project deleted');
+                });
+                this.set('preprintNode', null);
+                // TODO: reset dropzone, since uploaded file has no project
+            }
+            resolve();
+        },
+        preUpload(ignore, dropzone, file) {
+            this.set('uploadFile', file);
+            return new Ember.RSVP.Promise(resolve => this.set('resolve', resolve));
+        },
+        startUpload(resolve) {
+            this.set('_url', `${this.get('preprintNode.files').findBy('name', 'osfstorage').get('links.upload')}?kind=file&name=${this.get('uploadFile.name')}`);
+            this.get('resolve')();
+            this.get('toast').info('File will upload in the background.');
+            resolve();
+        },
+        buildUrl() {
+            return this.get('_url');
+        },
+        success() {
+            this.get('toast').info('File uploaded!');
+        },
+        deleteSubject(key, array = key.split('.')) {
+            this.set(key, null);
+            // Delete key manually
+            switch (array.length) {
+                case 2:
+                    delete this[array[0]][array[1]];
```

Comment:
It's not, but @hmoco was going to work on loading live data (it will also need to load incrementally, and search on the server)